sopel/tools/__init__.py

```diff
@@ -214,9 +214,10 @@ def compile_rule(nick, pattern, alias_nicks):
         nick = re.escape(nick)
 
     pattern = pattern.replace('$nickname', nick)
-    pattern = pattern.replace('$nick', r'{}[,:]\s+'.format(nick))
+    pattern = pattern.replace('$nick ', r'{}[,:]\s*'.format(nick))  # @rule('$nick hi')
+    pattern = pattern.replace('$nick', r'{}[,:]\s+'.format(nick))  # @rule('$nickhi')
     flags = re.IGNORECASE
-    if '\n' in pattern:
+    if '\n' in pattern:  # TODO: Why do we have this? IRC lines can't contain \n
```

Comment:
```suggestion
    if '\n' in pattern:
        # handle verbose regular expression
        # see https://docs.python.org/3/library/re.html#re.VERBOSE
```

Reply:
I did something like this, but without applying the suggestion. Good link to have in there!